WordPress/src/main/java/org/wordpress/android/datasets/ReaderBlogTable.java

```diff
@@ -90,10 +92,36 @@ public static ReaderBlog getBlogInfo(long blogId, String blogUrl) {
         }
     }
 
-    public static String getBlogUrl(long blogId) {
-        String[] args = {Long.toString(blogId)};
-        return SqlUtils.stringForQuery(
-                ReaderDatabase.getReadableDb(), "SELECT blog_url FROM tbl_blog_info WHERE blog_id=?", args);
+    public static long getBlogIdFromUrl(String url) {
+        if (TextUtils.isEmpty(url)) {
+            return 0;
+        }
+        String normUrl = UrlUtils.normalizeUrl(url);
+        String[] args = {normUrl, normUrl};
+        return SqlUtils.longForQuery(ReaderDatabase.getReadableDb(),
+                "SELECT blog_id FROM tbl_blog_info WHERE feed_url=? OR blog_url=?",
```

Comment:
I added that just in case we could get the ID from the other URL column, but now that you point it out it does seem confusing. I'll update this to only check the appropriate URL column.